### PR TITLE
Trigger Templates

### PR DESCRIPTION
**Pull request by InfusOnWoW**

### CODE REVIEW

File: Options/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua [hunk 1]
```diff
@@ -679,7 +679,7 @@ local methods = {
         self.frame.description = {...};
     end,
     ["SetIcon"] = function(self, icon)
-        if(type(icon) == "string") then
+        if(type(icon) == "string" or type(icon) == "number") then
```
Comment (Stanzilla) [inline Options/AceGUI-Widgets/AceGUIWidget-WeakAurasDisplayButton.lua]: file needs a minor bump

File: Options/AceGUI-Widgets/AceGUIWidget-WeakAurasNewButton.lua [hunk 2]
```diff
@@ -39,7 +39,7 @@ local methods = {
         self.frame:SetScript("OnClick", func);
     end,
     ["SetIcon"] = function(self, icon)
-        if(type(icon) == "string") then
+        if(type(icon) == "string" or type(icon) == "number") then
```
Comment (Stanzilla) [inline Options/AceGUI-Widgets/AceGUIWidget-WeakAurasNewButton.lua]: file needs a minor bump

File: Options/WeakAurasOptions.lua [hunk 3]
```diff
@@ -7375,15 +7428,6 @@ function WeakAuras.CreateFrame()
   codereviewCancel:SetWidth(100);
   codereviewCancel:SetText(L["Okay"]);
 
-  function codereview.Close(self)
```
Comment (Stanzilla) [inline Options/WeakAurasOptions.lua]: we don't need that anymore because of newView, right?

File: Templates/WeakAurasTemplates.toc [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+## Title: WeakAuras Templates
+## Author: Infus
+## Version: @project-version@
+## Notes: Templates for WeakAuras
+## X-Category: Interface Enhancements
+## DefaultState: Enabled
+## LoadOnDemand: 1
+## Dependencies: WeakAuras, WeakAurasOptions
+
+Localization-enUS.lua
```
Comment (Stanzilla) [inline Templates/WeakAurasTemplates.toc]: locale files are missing atm, just copy paste them over from Tutorials and replace `Tutorials` with `Templates` in them, I will hook it up with the translation system then

**Comment by Stanzilla** [comment]
Needs a rebase to get rid of the two extra commits

**Comment by Stanzilla** [comment]
Definitely needs a "back" button in the navigation

File: Options/WeakAurasOptions.lua (hunk 3 above)
Comment (InfusOnWoW) [inline Options/WeakAurasOptions.lua]: For some reason there were two codereview.Close() functions. There should be one left just a few lines later.

Comment (Stanzilla) [inline Options/WeakAurasOptions.lua]: Ah ok, makes sense. It should behave the same now when the second one already came after this one

**Comment by InfusOnWoW** [comment]
So I agree on all of those comments.

I'm not sure where to put the druid shapeshift trigger though. That's why the section was called Other. But yeah Ressources is better for all the other classes. I don't think the Shapeshift trigger fit into either Buffs/Debuffs or Cooldowns too well. So for now for Druids the secion is called Ressources and Shape Shift Form.

**Pull request merged**